### PR TITLE
[THOG-793] - Return all unverified results

### PR DESCRIPTION
**Pull request by ahrav**
Screen Shot 2022-10-30 at 1 32 43 PM
Remove the check to filter and return only a single unverified result.

https://github.com/trufflesecurity/trufflehog/issues/844

### CODE REVIEW

**Comment by trufflesteeeve** [comment]
This feels like a change that should be behind a CLI flag if we want to do it. Though `CleanResults`'s comment explains what it does, it doesn't explain why we're doing it - which is essentially to cut down on the noise of detecting secrets that have either already been rotated, or aren't real.

**Comment by ahrav** [comment]
> This feels like a change that should be behind a CLI flag if we want to do it. Though `CleanResults`'s comment explains what it does, it doesn't explain why we're doing it - which is essentially to cut down on the noise of detecting secrets that have either already been rotated, or aren't real.

Yea I agree. So this was actually a common thread in the community slack where users were not getting all the unverified secrets. After talking to Dustin we mainly did this before we had the Dashboard to filter out verified/ non-verified. So with that change in place this was the suggested solution since this would technically be the approach where there we no longer filter out actual unverified results.

That being said I think adding a cmd line flag for this makes sense 👍

**Comment by dustin-decker** [comment]
It was put in place before we grouped credentials, but we do now, so maybe it is no longer needed? It definitely causes some unexpected behavior.

**Comment by trufflesteeeve** [comment]
That seems reasonable. We can always add  functionality back in later if we need it. We might want to look into using more of [the AWS style custom cleaner](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/aws/aws.go#L182) for any key/secret detector types, especially if their secrets also match a standard hash.

**Comment by ahrav** [comment]
> That seems reasonable. We can always add functionality back in later if we need it. We might want to look into using more of [the AWS style custom cleaner](https://github.com/trufflesecurity/trufflehog/blob/main/pkg/detectors/aws/aws.go#L182) for any key/secret detector types, especially if their secrets also match a standard hash.

I have actually not seen this before. Yea I like this idea.

**Comment by ahrav** [comment]
OKay, I think I actually found a pretty nice solution here. I move out the call to `CleanResults` into the engine vs within each detector. I also then included a CLI flag if we want to filter unverified results and pass that into the Engine during construction.

File: README.md [hunk 1]
```diff
@@ -180,6 +180,7 @@ Flags:
       --concurrency=1            Number of concurrent workers.
       --no-verification          Don't verify the results.
       --only-verified            Only output verified results.
+      --filter-unverified        Only output first unverified result per chunk per detector if there are more than one results.
```
Comment (trufflesteeeve) [inline README.md:183]: __nitpick__: Typo - "more than one results." -> "more than one result."

File: main.go [hunk 2]
```diff
@@ -30,15 +30,16 @@ import (
...
+	cli              = kingpin.New("TruffleHog", "TruffleHog is a tool for finding credentials.")
+	cmd              string
+	debug            = cli.Flag("debug", "Run in debug mode.").Bool()
+	trace            = cli.Flag("trace", "Run in trace mode.").Bool()
+	jsonOut          = cli.Flag("json", "Output in JSON format.").Short('j').Bool()
+	jsonLegacy       = cli.Flag("json-legacy", "Use the pre-v3.0 JSON format. Only works with git, gitlab, and github sources.").Bool()
+	concurrency      = cli.Flag("concurrency", "Number of concurrent workers.").Default(strconv.Itoa(runtime.NumCPU())).Int()
+	noVerification   = cli.Flag("no-verification", "Don't verify the results.").Bool()
+	onlyVerified     = cli.Flag("only-verified", "Only output verified results.").Bool()
+	filterUnverified = cli.Flag("filter-unverified", "Only output first unverified result per chunk per detector if there are more than one results.").Bool()
```
Comment (trufflesteeeve) [inline main.go:42]: __nitpick__: Typo - "more than one results." -> "more than one result."

**Review by dustin-decker — APPROVED** [review]

**Review by trufflesteeeve — APPROVED** [review]
My only comments are about a couple of typos. Looks good!

**Pull request merged**